hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -488,64 +491,61 @@ func (r *HostedClusterReconciler) listNodePools(clusterNamespace, clusterName st
 	return filtered, nil
 }
 
-func (r *HostedClusterReconciler) delete(ctx context.Context, req ctrl.Request) error {
+func (r *HostedClusterReconciler) delete(ctx context.Context, req ctrl.Request) (*ctrl.Result, error) {
 	targetNamespace := req.Name
 
+	var result *ctrl.Result
+	result = nil
 	nodePools, err := r.listNodePools(req.Namespace, req.Name)
 	if err != nil {
-		return fmt.Errorf("failed to get nodePools by cluster name for cluster %q: %w", req.Name, err)
+		return result, fmt.Errorf("failed to get nodePools by cluster name for cluster %q: %w", req.Name, err)
 	}
 
 	for key := range nodePools {
 		if err := r.Delete(ctx, &nodePools[key]); err != nil && !apierrors.IsNotFound(err) {
-			return fmt.Errorf("failed to delete nodePool %q for cluster %q: %w", nodePools[key].GetName(), req.Name, err)
+			return result, fmt.Errorf("failed to delete nodePool %q for cluster %q: %w", nodePools[key].GetName(), req.Name, err)
 		}
 	}
 
-	r.Log.Info("Deleting cluster", "name", req.Name, "namespace", targetNamespace)
+	r.Log.Info("Deleting Cluster", "clusterName", req.Name, "clusterNamespace", targetNamespace)
 	cluster := &capiv1.Cluster{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      req.Name,
 			Namespace: targetNamespace,
 		},
 	}
 
-	if err := waitForDeletion(ctx, r.Log, r, cluster); err != nil && !apierrors.IsNotFound(err) {
-		return fmt.Errorf("failed to delete cluster: %w", err)
+	deleted, err := ensureClusterDeleted(ctx, r.Log, r, cluster)
```

Comment:
I like this, but for clarity of reading I would suggest just inlining all the logic of `ensureClusterDeleted` here since it's only a few lines and has no other callers. Not sure the indirection of the function adds anything.

Reply:
sure, as i was iterating on this what once was `waitForDeletion` did get smaller and smaller